api/_db.data-utils.js

```diff
@@ -0,0 +1,127 @@
+import { inspect } from 'util';
+import { strict as assert } from 'assert';
+
+const CODE = {
+	GREEN: 'green',
+	RED: 'red',
+	INSTRUCTOR: 'instructor',
+	NONE: 'none',
+	UNKNOWN: 'unknown',
+};
+
+const RAW_VALUE_TO_CODE = {
+	roheline: CODE.GREEN,
+	punane: CODE.RED,
+	instruktor: CODE.INSTRUCTOR,
+	'': CODE.NONE
+};
+
+// raw input from the sheet => valueof CODE
+const normalizeCertificate = (rawCertificate) => {
+	assert.equal(typeof rawCertificate, 'string', `Expected "rawCertificate" to be a string, got ${inspect(rawCertificate)}`);
+	const certificate = RAW_VALUE_TO_CODE[rawCertificate.toLowerCase()];
+
+	if (certificate) {
+		return certificate;
+	}
+
+	console.error(`Invalid certificate input: ${rawCertificate}`);
+	return CODE.UNKNOWN;
+};
+
+const assertValidDate = (parsed) => {
+	assert(parsed instanceof Date && !isNaN(parsed.valueOf()), `Invalid Date: "${inspect(parsed)}"`);
+};
+
+const formatDate = (parsed) => {
+	if (!parsed) {
+		return null;
+	}
+	assertValidDate(parsed);
+	return parsed.toISOString().substr(0, 10);
+};
+
+const parseDate = (rawValue) => {
+	if (!rawValue) {
+		return null;
+	}
+	const parsed = new Date(rawValue);
+	assert(!isNaN(parsed), `Invalid date: "${inspect(rawValue)}"`);
+	return parsed;
+};
+
+const S = 1000;
+const MIN = 60 * S;
+const HR = 60 * MIN;
+const D = 24 * HR;
+const getExpiryTimeFromFormFillTime = (normDate) => {
+	if (!normDate) {
+		return null;
+	}
+	assertValidDate(normDate);
+	// Add 6 weeks
+	const exp = normDate.valueOf() + 6 * 7 * D;
+	return new Date(exp);
+};
+
+// have to either map by the position or header name, doing the latter
+const filterColumnHeader = 'id';
+const certificateHeader = 'certificate';
+const nameHeader = 'name';
+const examinerHeader = 'examiner';
+const examDateHeader = 'examDate';
+const expiryDateHeader = 'expiryDate';
+const formFillTimeHeader = 'formFillTime';
+
+export const findById = (data, id) => {
+	// Drop human-readable headers
```

Comment:
This comment could elaborate a bit futher on why it's done like this.

Reply:
Updated the comment.